### ISSUE TITLE
predictions with nnet::multinom models not giving confidence intervals with type="response" & give error with type="latent"

### ISSUE THREAD

**Issue by tomwenseleers** [vincentarelbundock/marginaleffects#473]
Thanks for implementing `type="latent"` in `nnet::multinom` models. `get_predict.multinom` with `type="response"` and `type="latent"` are now working as expected. 

However, predictions() with `nnet::multinom` do not return confidence intervals with type="response", and return an error with type="latent".

Example:
```
library(marginaleffects)
library(nnet)
library(emmeans)
library(splines)

# Dataset: covid variant frequencies ("variant") through time ("collection_date_num") ####
# "count" = actual count per week
dat = read.csv("https://www.dropbox.com/s/u27cn44p5srievq/dat.csv?dl=1")
# simplify example to just 3 categories:
dat = dat[dat$variant %in% c("Alpha", "Beta", "Other"),] 
dat$variant = factor(dat$variant, levels=c("Other", "Beta", "Alpha"))

set.seed(1)
fit_multinom = nnet::multinom(variant ~ ns(collection_date_num, df=2), 
                          weights=count, data=dat)
```

`predictions() `with `type="response"` does not return confidence intervals:
```
predictions(fit_multinom, 
            newdata = datagrid(collection_date_num=dat[160,'collection_date_num']),
            type = "probs")
#   rowid  type group  predicted   std.error    count collection_date_num variant
# 1     1 probs Other 0.86628084 0.007356969 73.10398               18764   Alpha
# 2     1 probs  Beta 0.01791583 0.002713518 73.10398               18764   Alpha
# 3     1 probs Alpha 0.11580333 0.006993107 73.10398               18764   Alpha
```

`predictions()` with` type="latent" `returns an error:
```
predictions(fit_multinom, 
            newdata = datagrid(collection_date_num=dat[160,'collection_date_num']),
            type = "latent")
# Error in rowMeans(log_mu) : 
#   'x' must be an array of at least two dimensions
```

The expected result (obtained using `emmeans`) for these two calls would be
```
# emmeans result (response/prob scale, with SEs and 95% CLs on response scale)
as.data.frame(emmeans(fit_multinom, ~ collection_date_num, by="variant", 
                      at=list(collection_date_num=dat[160,'collection_date_num']),
                      mode="prob", level=0.95))
#   collection_date_num variant       prob          SE df   lower.CL  upper.CL
# 1               18764   Other 0.86628084 0.007354173  6 0.84828582 0.8842759
# 2               18764    Beta 0.01791583 0.002712467  6 0.01127867 0.0245530
# 3               18764   Alpha 0.11580333 0.006990414  6 0.09869840 0.1329083

# emmeans result (latent scale, with SEs and 95% CLs on latent scale)
as.data.frame(emmeans(fit_multinom, ~ collection_date_num, by="variant", 
                      at=list(collection_date_num=dat[160,'collection_date_num']),
                      mode="latent", level=0.95))
#   collection_date_num variant      emmean         SE df  lower.CL   upper.CL
# 1               18764   Other  1.96361336 0.05710552  6  1.823881  2.1033455
# 2               18764    Beta -1.91491087 0.10442145  6 -2.170421 -1.6594008
# 3               18764   Alpha -0.04870248 0.06727033  6 -0.213307  0.1159021
```

NOTE: for `mclogit::mblogit `multinomial models everything works as expected, and adjusted predictions are way faster than in `emmeans` in fact (especially when working on the response scale), so that's great (only `emtrends `are a bit slower); I think that in the end using `type="latent"` followed by a softmax backtransform is the logical thing to do (in the same way that for adjusted predictions of a binomial GLM `type="link"` followed by an inverse logit backtransform feels most logical to me) :
```
# timings adjusted predictions emmeans vs marginaleffects
dates = seq(min(dat$collection_date_num), max(dat$collection_date_num)) 
system.time(as.data.frame(emmeans(fit_mblogit, ~ collection_date_num, by="variant", 
                      at=list(collection_date_num=dates),
                      mode="prob", level=0.95))) # emmeans response scale 22s
system.time(as.data.frame(emmeans(fit_mblogit, ~ collection_date_num, by="variant", 
                                  at=list(collection_date_num=dates),
                                  mode="latent", level=0.95))) # emmeans latent scale 0.19s
system.time(predictions(fit_mblogit, 
            newdata = datagrid(collection_date_num=dates),
            type="response")) # marginaleffects response scale 0.03s
system.time(predictions(fit_mblogit, 
                        newdata = datagrid(collection_date_num=dates),
                        type="latent")) # marginaleffects latent scale 0.09s

# timings emtrends vs emtrends in marginaleffects
library(microbenchmark)
microbenchmark(as.data.frame(confint(pairs(emtrends(fit_mblogit, ~ variant|collection_date_num,
                                     var = "collection_date_num",  
                                     mode = "latent",
                                     at = list(collection_date_num = 
                                                 dat[160,'collection_date_num'])),
                            reverse=TRUE), level=0.9))) # emtrends 83 ms

microbenchmark(comparisons(
  fit_mblogit,
  newdata = datagrid(collection_date_num = 
                       c(dat[160,'collection_date_num'])),
  variables = "collection_date_num",
  type = "latent",
  hypothesis = "pairwise")) # marginaleffects 245 ms

```

**Comment by vincentarelbundock** [comment]
> `predictions()` with`type="latent"`returns an error:

Great catch! The bug was because the `predict()` function returns a vector instead of a matrix when the input `newdata` has only one row. I pushed bugfix and things should work now.

> However, predictions() with `nnet::multinom` do not return confidence intervals with type="response", and return an error with type="latent".

We return confidence intervals only for (1) linear models and (2) models for which we can predict on the link scale and backtransform to get proper intervals. In other cases, we supply a delta method standard errors, but we do not build symmetric intervals, because those may not always make sense. Of course, users can easily do:

```r
predictions(mod) |> 
    transform(conf.low = estimate - 1.96 * std.error,
              conf.high = estimate + 1.96 * std.error)
```

>  adjusted predictions are way faster than in `emmeans` in fact (especially when working on the response scale), so that's great (only `emtrends `are a bit slower);

I am very (pleasantly) surprised by this. Excellent news! Thanks for testing this out.
